### PR TITLE
Complete core tests for CultureData Unix

### PR DESCRIPTION
**Pull request by steveharter**
All code review comments have been replied to and code updated. Please take a look.

### CODE REVIEW

File: src/System.Globalization/tests/CultureInfo/CultureInfoDateTimeFormat.cs [hunk 1]
```diff
@@ -37,6 +38,44 @@ public void PosTest2()
...
+        [Theory]
+        [InlineData("en-US", "h:mm tt")]
+        public void TestShortTimePatternLocale(string locale, string expected)
+        {
+            CultureInfo myTestCulture = new CultureInfo(locale);
+            DateTimeFormatInfo dtf = myTestCulture.DateTimeFormat;
+            Assert.Equal(expected, dtf.ShortTimePattern);
+        }
+
+        [InlineData("fi-FI", "H:mm", "H.mm")]
```
Comment (eerhardt) [inline src/System.Globalization/tests/CultureInfo/CultureInfoDateTimeFormat.cs:51]: Does this method need a [Theory] attribute on it?

File: src/System.Globalization/tests/CultureInfo/CultureInfoDateTimeFormat.cs [hunk 2]
```diff
@@ -37,6 +38,44 @@ public void PosTest2()
...
+        [Theory]
+        [InlineData("en-US", "h:mm:ss tt")]
+        public void TestLongTimePatternLocale(string locale, string expected)
+        {
+            CultureInfo myTestCulture = new CultureInfo(locale);
+            DateTimeFormatInfo dtf = myTestCulture.DateTimeFormat;
+            Assert.Equal(expected, dtf.LongTimePattern);
+        }
+
+        [InlineData("fi-FI", "H:mm:ss", "H.mm.ss")]
```
Comment (eerhardt) [inline src/System.Globalization/tests/CultureInfo/CultureInfoDateTimeFormat.cs:71]: Does this method need a [Theory] attribute on it?

**Comment by ellismg** [comment]
Thanks @steveharter!  I had one small question about why a test is now failing.

File: src/System.Globalization/tests/CultureInfo/CultureInfoEnglishName.cs [hunk 3]
```diff
@@ -18,5 +18,13 @@ public void PosTest1()
             string excepectedName = new CultureInfo(ci.Name).EnglishName;
             Assert.Equal(excepectedName, inFactenglish);
         }
+
+        [Theory]
+        [InlineData("en-US", "English (United States)")]
```
Comment (eerhardt) [inline src/System.Globalization/tests/CultureInfo/CultureInfoEnglishName.cs:23]: Can we add 1 or 2 more cultures here?

File: src/System.Globalization/tests/CultureInfo/CultureInfoNativeName.cs [hunk 4]
```diff
@@ -19,5 +19,13 @@ public void PosTest1()
                 Assert.Equal(excepectedName, inFactName);
             }
         }
+
+        [Theory]
+        [InlineData("en-US", "English (United States)")]
+        public void TestNativeNameLocale(string locale, string expected)
```
Comment (eerhardt) [inline src/System.Globalization/tests/CultureInfo/CultureInfoNativeName.cs:26]: Can we add 1 or 2 more cultures here?

File: src/System.Globalization/tests/CultureInfo/CultureInfoDateTimeFormat.cs (hunk 1 above)
Comment (ellismg) [inline src/System.Globalization/tests/CultureInfo/CultureInfoDateTimeFormat.cs:51]: I would be fine keeping it as is, so we could easily add additional locale cases (the difference between this and the next one is just having a way to signal that ICU data differs)

File: src/System.Globalization/tests/NumberFormatInfo/NumberFormatInfoNumberGroupSizes.cs [hunk 5]
```diff
@@ -0,0 +1,95 @@
...
+            NumberFormatInfo nfi = new NumberFormatInfo();
+            nfi.NumberGroupSizes = new int[] { 2, 3, 4 };
+            int[] expected = nfi.NumberGroupSizes;
+            Assert.Equal(3, expected.Length);
+            Assert.Equal(2, expected[0]);
+            Assert.Equal(3, expected[1]);
+            Assert.Equal(4, expected[2]);
+        }
+
+        // NegTest1: ArgumentNullException is not thrown
```
Comment (eerhardt) [inline src/System.Globalization/tests/NumberFormatInfo/NumberFormatInfoNumberGroupSizes.cs]: This comment says "is not thrown", but the tests asserts that it does throw.

Comment (eerhardt) [inline src/System.Globalization/tests/NumberFormatInfo/NumberFormatInfoNumberGroupSizes.cs]: Repeated in the tests below.

File: src/System.Globalization/tests/CultureInfo/CultureInfoDateTimeFormat.cs (hunk 1 above)
Comment (eerhardt) [inline src/System.Globalization/tests/CultureInfo/CultureInfoDateTimeFormat.cs:51]: My point is that this test doesn't even run right now, right?  I thought if there is no [Fact] or [Theory] attribute, then xUnit doesn't even execute the method.

Comment (ellismg) [inline src/System.Globalization/tests/CultureInfo/CultureInfoDateTimeFormat.cs:51]: Sorry, misunderstood the comment.  Yes, it should have a [Theory].  Thought you were saying a [Theory] with only one piece of data could be a [Fact],

Comment (steveharter) [inline src/System.Globalization/tests/CultureInfo/CultureInfoDateTimeFormat.cs:51]: Fixed - added attribute

File: src/System.Globalization/tests/CultureInfo/CultureInfoEnglishName.cs (hunk 3 above)
Comment (steveharter) [inline src/System.Globalization/tests/CultureInfo/CultureInfoEnglishName.cs:23]: Sure added fr-FR

File: src/System.Globalization/tests/CultureInfo/CultureInfoNativeName.cs (hunk 4 above)
Comment (steveharter) [inline src/System.Globalization/tests/CultureInfo/CultureInfoNativeName.cs:26]: Sure added en-CA

File: src/System.Globalization/tests/NumberFormatInfo/NumberFormatInfoNumberGroupSizes.cs (hunk 5 above)
Comment (steveharter) [inline src/System.Globalization/tests/NumberFormatInfo/NumberFormatInfoNumberGroupSizes.cs]: fixed

File: src/System.Globalization/tests/CultureInfo/CultureInfoDateTimeFormat.cs (hunk 2 above)
Comment (steveharter) [inline src/System.Globalization/tests/CultureInfo/CultureInfoDateTimeFormat.cs:71]: Fixed; good catch

**Comment by eerhardt** [comment]
LGTM, assuming the merge conflicts are trivial.

**Comment by ellismg** [comment]
I addressed the merge conflicts locally and pushed the changes into the branch.  Thanks again, Steve!

**Pull request merged**